middleware/proxy/proxy.go

```diff
@@ -57,6 +58,16 @@ func (uh *UpstreamHost) Down() bool {
 	return uh.CheckDown(uh)
 }
 
+// Full checks whether the upstream host has reached its maximum connections
+func (uh *UpstreamHost) Full() bool {
+	return uh.MaxConns > 0 && uh.Conns >= uh.MaxConns
```

Comment:
sorry, uh.Conns, since it's modified atomically as requests come and go in various goroutines.  I'll submit a PR shortly.